anfisa.filipova/task-2-2/cmd/service/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"container/heap"
+	"errors"
+	"fmt"
+
+	"github.com/Anfisa111/task-2-2/internal/intheap"
+)
+
+var (
+	errReading          = errors.New("error reading")
+	errTypeAssertion    = errors.New("error type assertion")
+	errInvalidChoiceIdx = errors.New("error invalid choiceIdx")
+)
+
+func readIndex() (int, error) {
+	var choiceIdx int
+	if _, err := fmt.Scan(&choiceIdx); err != nil {
+		return 0, fmt.Errorf("%w: %w", errReading, err)
+	}
+
+	return choiceIdx, nil
+}
+
+func readDishes() (intheap.IntHeap, error) {
+	var dishCount int
+	if _, err := fmt.Scan(&dishCount); err != nil {
+		return nil, fmt.Errorf("%w: %w", errReading, err)
+	}
+
+	heapint := &intheap.IntHeap{}
+
+	heap.Init(heapint)
+
+	for range dishCount {
+		var priority int
+		if _, err := fmt.Scan(&priority); err != nil {
+			return nil, fmt.Errorf("%w: %w", errReading, err)
```

Comment:
зачем тут станическая ошибка?

```suggestion
			return nil, fmt.Errorf("failed to read priority: %w", errReading, err)
```

все еще нет контекста для разных ошибок + нужно поменять статические ошибки, раз ты их ниже не производишь приведение типов